ibm/service/satellite/data_source_ibm_satellite_host_script.go

```diff
@@ -188,30 +194,34 @@ elif [[ "${OPERATING_SYSTEM}" == "RHEL8" ]]; then
 	subscription-manager repos --enable rhel-8-for-x86_64-baseos-rpms 
 	subscription-manager repos --enable rhel-8-for-x86_64-appstream-rpms;
 fi
-yum install container-selinux -y`
-					} else if strings.ToLower(hostProvider) == "azure" {
-						lines[i] = `
+yum install container-selinux -y
+`
+		case strings.ToLower(hostProvider) == "azure":
+			insertionText = `
 if [[ "${OPERATING_SYSTEM}" == "RHEL8" ]]; then
 	update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.8 1
 	update-alternatives --set python3 /usr/bin/python3.8
 fi
 yum install container-selinux -y
 `
-					} else if strings.ToLower(hostProvider) == "google" {
-						lines[i] = `
+		case strings.ToLower(hostProvider) == "google":
+			insertionText = `
 if [[ "${OPERATING_SYSTEM}" == "RHEL8" ]]; then
 	update-alternatives --install /usr/bin/python3 python3 /usr/bin/python3.8 1
 	update-alternatives --set python3 /usr/bin/python3.8
 fi
 yum install container-selinux -y
 `
-					}
-				}
+		default:
+			if script, ok := d.GetOk("custom_script"); ok {
+				insertionText = script.(string)
 			}
 		}
+
+		lines[index] = lines[index] + "\n" + insertionText
```

Comment:
I think here we might actually want insertionText -> lines[index] instead of the reverse. Currently we look for `operating_system` export, and want our code to execute immediately after.  If there was 
```
export OPERATING_SYSTEM <- insert after here
if A
then B
```

we would end. up with 

```
export OPERATING_SYSTEM
if A <- index = line for operating system + 1
insertion text
then B
```

Reply:
In my test it inserted as expected:

```
export OPERATING_SYSTEM
subscription-manager refresh
# subscription-manager repos --enable rhel-server-rhscl-7-rpms
# subscription-manager repos --enable rhel-7-server-optional-rpms
# subscription-manager repos --enable rhel-7-server-rh-common-rpms
# subscription-manager repos --enable rhel-7-server-supplementary-rpms
# subscription-manager repos --enable rhel-7-server-extras-rpms
set -x
subscription-manager repos --enable rhel-8-for-x86_64-appstream-rpms
subscription-manager repos --enable rhel-8-for-x86_64-baseos-rpms

<snip>

systemctl enable $SERVICENAME


if [[ "${OPERATING_SYSTEM}" != "RHEL7" && "${OPERATING_SYSTEM}" != "RHEL8" ]]; then
	echo "This script is only intended to run with a RHEL7 or RHEL8 operating system. Current operating system ${OPERATING_SYSTEM}."
	exit 1
fi
```


which aligns with when we break - we break and index the line that `export Operating_System` is on:

```
		var index int
 		for i, line := range lines {
 			if strings.Contains(line, `export OPERATING_SYSTEM`) {
 				index = i
 				break
 			}
 ```

Reply:
(that is a change from the original code...there had been an index + 1)